site/src/xServices/workspaces/workspacesXService.ts

```diff
@@ -1,64 +1,263 @@
 
-interface WorkspaceContext {
-  workspaces?: TypesGen.Workspace[]
+/**
+ * Workspace item machine
+ *
+ * It is used to control the state and actions of each workspace in the
+ * workspaces page view
+ **/
+interface WorkspaceItemContext {
+  data: TypesGen.Workspace
+  updatedTemplate?: TypesGen.Template
+}
+
+type WorkspaceItemEvent =
+  | {
+      type: "UPDATE_VERSION"
+    }
+  | {
+      type: "UPDATE_DATA"
+      data: TypesGen.Workspace
+    }
+
+export const workspaceItemMachine = createMachine(
+  {
+    id: "workspaceItemMachine",
+    schema: {
+      context: {} as WorkspaceItemContext,
+      events: {} as WorkspaceItemEvent,
+      services: {} as {
+        getTemplate: {
+          data: TypesGen.Template
+        }
+        startWorkspace: {
+          data: TypesGen.WorkspaceBuild
+        }
+        getWorkspace: {
+          data: TypesGen.Workspace
+        }
+      },
+    },
+    tsTypes: {} as import("./workspacesXService.typegen").Typegen0,
+    type: "parallel",
+
+    states: {
+      updateVersion: {
+        initial: "idle",
+        states: {
+          idle: {
+            on: {
+              UPDATE_VERSION: {
+                target: "gettingUpdatedTemplate",
+                // We can improve the UI by optimistically updating the workspace status
+                // to "Queued" so the UI can display the updated state right away and we
+                // don't need to display an extra spinner.
+                actions: ["assignQueuedStatus", "displayUpdatingVersionMessage"],
+              },
+              UPDATE_DATA: {
+                actions: "assignUpdatedData",
+              },
+            },
+          },
+          gettingUpdatedTemplate: {
+            invoke: {
+              id: "getTemplate",
+              src: "getTemplate",
+              onDone: {
+                actions: "assignUpdatedTemplate",
+                target: "restartingWorkspace",
+              },
+              onError: {
+                target: "idle",
+                actions: "displayUpdateVersionError",
+              },
+            },
+          },
+          restartingWorkspace: {
+            invoke: {
+              id: "startWorkspace",
+              src: "startWorkspace",
+              onDone: {
+                actions: "assignLatestBuild",
+                target: "waitingToBeUpdated",
+              },
+              onError: {
+                target: "idle",
+                actions: "displayUpdateVersionError",
+              },
+            },
+          },
+          waitingToBeUpdated: {
+            after: {
+              5000: "gettingUpdatedWorkspaceData",
+            },
+          },
+          gettingUpdatedWorkspaceData: {
+            invoke: {
+              id: "getWorkspace",
+              src: "getWorkspace",
+              onDone: [
+                {
+                  target: "waitingToBeUpdated",
+                  cond: "isOutdated",
```

Comment:
New to XState so asking some basic questions: does this `cond` apply to the target? Like we won't poll for updates unless the workspace is outdated?